##                               ANSWER OF TASK1

1.1.Download the data set movie_metadata.csv,which contains data about films from IMDb

In [96]:
#1-1import  pandas and numpy library 
import pandas as pd
import numpy as np
file= r"C:\Users\bc\Documents\test\movie_metadata.csv"#donnez le repertoire du fichier
movie_metadata= pd.read_csv(file)

In [97]:
movie_metadata

color      director_name  num_critic_for_reviews  duration  \
0     Color      James Cameron                   723.0     178.0   
1     Color     Gore Verbinski                   302.0     169.0   
2     Color         Sam Mendes                   602.0     148.0   
3     Color  Christopher Nolan                   813.0     164.0   
4       NaN        Doug Walker                     NaN       NaN   
...     ...                ...                     ...       ...   
5038  Color        Scott Smith                     1.0      87.0   
5039  Color                NaN                    43.0      43.0   
5040  Color   Benjamin Roberds                    13.0      76.0   
5041  Color        Daniel Hsia                    14.0     100.0   
5042  Color           Jon Gunn                    43.0      90.0   

      director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                         0.0                   855.0  Joel David Moore   
1                       563.0                  1000.0     Orlando Bloom   
2                         0.0                   161.0      Rory Kinnear   
3                     22000.0                 23000.0    Christian Bale   
4                       131.0                     NaN        Rob Walker   
...                       ...                     ...               ...   
5038                      2.0                   318.0     Daphne Zuniga   
5039                      NaN                   319.0     Valorie Curry   
5040                      0.0                     0.0     Maxwell Moody   
5041                      0.0                   489.0     Daniel Henney   
5042                     16.0                    16.0  Brian Herzlinger   

      actor_1_facebook_likes        gross                           genres  \
0                     1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi   
1                    40000.0  309404152.0         Action|Adventure|Fantasy   
2                    11000.0  200074175.0        Action|Adventure|Thriller   
3                    27000.0  448130642.0                  Action|Thriller   
4                      131.0          NaN                      Documentary   
...                      ...          ...                              ...   
5038                   637.0          NaN                     Comedy|Drama   
5039                   841.0          NaN     Crime|Drama|Mystery|Thriller   
5040                     0.0          NaN            Drama|Horror|Thriller   
5041                   946.0      10443.0             Comedy|Drama|Romance   
5042                    86.0      85222.0                      Documentary   

      ... num_user_for_reviews language  country  content_rating  \
0     ...               3054.0  English      USA           PG-13   
1     ...               1238.0  English      USA           PG-13   
2     ...                994.0  English       UK           PG-13   
3     ...               2701.0  English      USA           PG-13   
4     ...                  NaN      NaN      NaN             NaN   
...   ...                  ...      ...      ...             ...   
5038  ...                  6.0  English   Canada             NaN   
5039  ...                359.0  English      USA           TV-14   
5040  ...                  3.0  English      USA             NaN   
5041  ...                  9.0  English      USA           PG-13   
5042  ...                 84.0  English      USA              PG   

            budget  title_year actor_2_facebook_likes imdb_score  \
0     237000000.0$      2009.0                  936.0        7.9   
1     300000000.0$      2007.0                 5000.0        7.1   
2     245000000.0$      2015.0                  393.0        6.8   
3     250000000.0$      2012.0                23000.0        8.5   
4             0.0$         NaN                   12.0        7.1   
...            ...         ...                    ...        ...   
5038          0.0$      2013.0                  470.0     

1.2-The duration column contains data on the film length. How many missing values are there in this column?

In [98]:
#1-2-nombre de valeurs manquantes dans la colonne "duration". La colonne duration a 15 valeurs manquantes
movie_metadata['duration'].isna().sum()

15

There are **15** missing values in column duration

In [99]:
movie_metadata.isna().sum()

color                         19
director_name                104
num_critic_for_reviews        50
duration                      15
director_facebook_likes      104
actor_3_facebook_likes        23
actor_2_name                  13
actor_1_facebook_likes         7
gross                        884
genres                         0
actor_1_name                   7
movie_title                    0
num_voted_users                0
cast_total_facebook_likes      0
actor_3_name                  23
facenumber_in_poster          13
plot_keywords                153
movie_imdb_link                0
num_user_for_reviews          21
language                      12
country                        5
content_rating               303
budget                         0
title_year                   108
actor_2_facebook_likes        13
imdb_score                     0
aspect_ratio                 329
movie_facebook_likes           0
dtype: int64

1.3 Replace the missing values in the duration column with the median value for this column

In [100]:
movie_metadata_no_nan=movie_metadata.fillna(value={'duration':movie_metadata["duration"].median()})
movie_metadata_no_nan
movie_metadata_no_nan['duration'].isna().sum()

0

Now, duration have 0 missing value. we are using the fillna function

1.4 what is the average film length?Give the answer as a floating-point figure rounded to two decimal

In [101]:
average_length= round(movie_metadata_no_nan['duration'].mean(),2)
average_length

107.19

Using the function mean, the average films length is **107.19**

1.5-create a movie_duration_category column, which will contain three category depending on the film length

In [102]:
nv_val =[]
for i in np.arange(len(movie_metadata_no_nan.index)):
    if movie_metadata_no_nan.duration[i]<90:
        nv_val.append("<90")
    elif movie_metadata_no_nan.duration[i]>=90 and movie_metadata_no_nan.duration[i]<=120:
        nv_val.append("90-120")
    else:
        nv_val.append(">120")

In [103]:
movie_metadata_no_nan_new=movie_metadata_no_nan.assign(movie_duration_category=nv_val)
movie_metadata_no_nan_new['movie_duration_category']

0         >120
1         >120
2         >120
3         >120
4       90-120
         ...  
5038       <90
5039       <90
5040       <90
5041    90-120
5042    90-120
Name: movie_duration_category, Length: 5043, dtype: object

1.6 Build a summary table for films released after 2000 (inclusively), to list the numbers of films

In [104]:
movie_metadata_no_nan_new["title_year"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 5043 entries, 0 to 5042
Series name: title_year
Non-Null Count  Dtype  
--------------  -----  
4935 non-null   float64
dtypes: float64(1)
memory usage: 39.5 KB


In [105]:
masque=movie_metadata_no_nan_new["title_year"]>2000
summary_movie=movie_metadata_no_nan_new[masque]
summary_movie.groupby(['title_year','movie_duration_category'])['movie_duration_category'].aggregate('count').unstack()

movie_duration_category  90-120  <90  >120
title_year                                
2001.0                      120   29    39
2002.0                      146   36    27
2003.0                      108   31    30
2004.0                      142   30    42
2005.0                      142   31    48
2006.0                      146   40    53
2007.0                      130   31    43
2008.0                      160   29    36
2009.0                      178   42    40
2010.0                      171   26    33
2011.0                      150   36    39
2012.0                      139   41    41
2013.0                      145   38    54
2014.0                      171   36    45
2015.0                      145   34    47
2016.0                       79    8    19

1.7 how many films between  90 minutes and two hours long  where released in 2008?

**160** films  are released betweeen 90 minutes and 2 hours

In [106]:
movie_metadata_no_nan_new["plot_keywords"]

0                  avatar|future|marine|native|paraplegic
1       goddess|marriage ceremony|marriage proposal|pi...
2                     bomb|espionage|sequel|spy|terrorist
3       deception|imprisonment|lawlessness|police offi...
4                                                     NaN
                              ...                        
5038               fraud|postal worker|prison|theft|trial
5039         cult|fbi|hideout|prison escape|serial killer
5040                                                  NaN
5041                                                  NaN
5042    actress name in title|crush|date|four word tit...
Name: plot_keywords, Length: 5043, dtype: object

1.8-Create a column called movie_plot_category,to contain four category depending on the key words in the column

In [107]:
nv_val2 =[]
for i in np.arange(len(movie_metadata_no_nan_new.index)):
    if "love" and "death" in str(movie_metadata_no_nan_new.plot_keywords[i]):
        nv_val2.append("love_and_death")
    elif "love" in  str(movie_metadata_no_nan_new.plot_keywords[i]):
        nv_val2.append("love")
    elif "death" in str(movie_metadata_no_nan_new.plot_keywords[i]):
        nv_val2.append("death")
    else:
        nv_val2.append("other")

In [108]:
movie_metadata_no_nan_new2=movie_metadata_no_nan_new.assign(movie_plot_category=nv_val2)
movie_metadata_no_nan_new2['movie_plot_category']

0       other
1       other
2       other
3       other
4       other
        ...  
5038    other
5039    other
5040    other
5041    other
5042    other
Name: movie_plot_category, Length: 5043, dtype: object

1.9-Build a table to reflect the average rating of films depending on which movie_plot_category they belong to.

In [109]:
round(movie_metadata_no_nan_new2.groupby(['movie_plot_category'])['imdb_score'].agg(['mean']),2)

mean
movie_plot_category      
love                 6.58
love_and_death       6.53
other                6.43

1.10-what is the avarage rating of film's budget  in  the love cathegory

The average rating of films in the "love" category is **6.58**

1.11 what is the median budget for all the films listed

In [110]:
t= int(movie_metadata_no_nan_new2["budget"].str.replace('\W',"").median())
t

C:\Users\bc\AppData\Local\Temp\ipykernel_10176\924720371.py:1: FutureWarning: The default value of regex will change from True to False in a future version.
  t= int(movie_metadata_no_nan_new2["budget"].str.replace('\W',"").median())


150000000

The median budget is **150 000 000$**

**3 hours were required to complete the entire exercise**